apollo-router/src/plugins/telemetry/mod.rs

```diff
@@ -1223,9 +1228,10 @@ impl Telemetry {
                                     latency: duration,
                                     has_errors,
                                     persisted_query_hit,
+                                    requests_without_field_instrumentation,
                                     ..Default::default()
```

Comment:
Are there any other fields here that need filling out?

Reply:
if i remove `..Default::default()`
```
apollo-router/src/plugins/telemetry/mod.rs:1227:54
missing structure fields:
- cache_hit
- cache_latency
- root_error_stats
- public_cache_ttl_latency
- private_cache_ttl_latency
- registered_operation
- forbidden_operation
```

Reply:
The full struct is:

```rust
// TODO Make some of these fields bool
#[derive(Default, Debug, Serialize)]
pub(crate) struct SingleQueryLatencyStats {
    pub(crate) latency: Duration,
    pub(crate) cache_hit: bool,
    pub(crate) persisted_query_hit: Option<bool>,
    pub(crate) cache_latency: Option<Duration>,
    pub(crate) root_error_stats: SinglePathErrorStats,
    pub(crate) has_errors: bool,
    pub(crate) public_cache_ttl_latency: Option<Duration>,
    pub(crate) private_cache_ttl_latency: Option<Duration>,
    pub(crate) registered_operation: bool,
    pub(crate) forbidden_operation: bool,
    pub(crate) requests_without_field_instrumentation: u64,
}

Reply:
@BrynCooke I’m a bit confused reading this code, to the point where I wonder if the code itself is confused too. Is `SingleQueryLatencyStats` meant to apply to a single **subgraph** query? Because we build it for each supergraph query, which can involve any number of subgraph queries (and as many blobs of subgraph FTV1 data).

Reply:
Knowing what I know about these fields, I would say it shouldn't only be applying to a single subgraph query.  In fact, I would say most if not all of these should be based on the "supergraph" (or precisely, client -> router).  There's potentially a couple ones which might be _informed_ by the subgraphs, but I can't even say definitively how that would be without diving in deeper.